### ISSUE TITLE
How are ecological and geographical distances computed?

### ISSUE THREAD

**Issue by LafontRapnouilTristan** [fitzLab-AL/gdm#45]
Hey, 

thank you for this very convenient package.
I am digging around trying to understand how it works but can't find how are the distances 
computed. 
What are the metrics used to compute ecological distance and can we change it and are the predictors scaled?

Regarding geographical distances I was wondering about the interpretation of the splines:
![gdm_filt_lt](https://github.com/user-attachments/assets/839c1a39-5f07-4b25-9fa1-0ffb1f4cd341)

For instance, in the above data set, my maximal distance between two site is 12000km but the scale
of "geographic" is stopping at ~250.
Are these kilometers? And if yes, does it mean that pairwise dissimilarity between sites increases up to 250km 
and then what?

I hope that I am clear enough and thank you for your work and time!
Cheers!

**Comment by fitzLab-AL** [comment]
Hi @LafontRapnouilTristan - Thanks for your message. If you haven't done so already, I recommend you read some of the background papers to understand how ecological distance is calculated, scaling of the predictors, and interpretation of the fitted I-splines.

Ferrier, S., Manion, G., Elith, J. and Richardson, K. (2007), Using generalized dissimilarity modelling to analyse and predict patterns of beta diversity in regional biodiversity assessment. Diversity and Distributions, 13: 252-264. https://doi.org/10.1111/j.1472-4642.2007.00341.x

Mokany, K., Ware C., Woolley S. N. C., Ferrier S., & Fitzpatrick M. C. (2022). A working guide to harnessing generalized dissimilarity modelling for biodiversity analysis and conservation assessment. Global Ecology and Biogeography, 31, 802–821. https://doi.org/10.1111/geb.13459 

Geographic distances are in units of the coordinates and are Euclidean (as opposed to great circle). If the coordinates are geographic (lon / lat), then the distances will be in those units, not meters (or km).

**Comment by LafontRapnouilTristan** [comment]
Hi,
Thanks for the answer. I will read the papers!
Regarding geography is it possible to use great circle distances instead of euclidean and get km?
Or is it just not supported by the package?
Thanks again!

**Comment by fitzLab-AL** [comment]
As currently implemented, the package calculates Euclidean distances only. If you want those distances in meters, then you need to use a projection for your coordiantes that is in meters (and ideally preserves distances between locations).

**Comment by LafontRapnouilTristan** [comment]
Hi, been a while!
Just to say that if I'm not mistaken, such transformation don't exist for worldwide studies. It would be a great feature if GDM could support non euclidean distances!

**Comment by jarioksa** [comment]
With current almost-dependencies (Suggests: terra), it would be trivial to implement great circle distances. However, if there are calculations that assume Euclidean geometry (triangle inequality), great circle distances *may* fail.

**Comment by fitzLab-AL** [comment]
One of the challenges is that geographic distance are calculated in the C++ code.

**Comment by jarioksa** [comment]
**terra** has function `terra::distance` which has a `data.frame` method. With argument `lonlat = TRUE` it will find great circle distances (Haversine or optionally more accurate). Naturally, it also has methods for **terra** spatial objects. From the user point of view, you only communicate with **R** although internally **terra** uses compiled code.

**Comment by fitzLab-AL** [comment]
I am aware of `terra::distance` function. I was unclear in my previous comment. The C++ code I referred to is that used in the `gdm` package. Presumably one could calculate great circle distances and then pass those along to `gdm's` C++ code, but I am not a C++ programmer myself.

**Comment by jarioksa** [comment]
Yes, I see (now) what you mean. Implementing a simplified version of great circle distances is not too complicated. It is just a bit longer equation than the Euclidean you have now. With 'simplified' I mean the equation that assumes earth is a sphere (which is not a bad approximation) and then getting the distances in angular units which may be transformed to metric (angular minute is equal to nautical mile) assuming earth is spherical. I spotted the place in your C++ code where this should be implemented and could do it there "easily", but the real problem is to first learn the complex structure of your C++ library to understand if this needs changes in other parts of the code.

**Comment by fitzLab-AL** [comment]
This has been on my to-do list for too long now. Hopefully I can find some time in the new year to dig into it & implement a solution. Many thanks for your thoughts!
